programs/bpf_loader/src/syscalls.rs

```diff
@@ -1144,765 +1382,1727 @@ impl<'a> SyscallObject<BpfError> for SyscallKeccak256<'a> {
-        let invoke_context = self.invoke_context.borrow();
-        let enforce_aligned_host_addrs =
-            invoke_context.is_feature_active(&enforce_aligned_host_addrs::id());
+        result: &mut Result<u64, EbpfError<BpfError>>,
+    ) {
+        // Get the 3 BigNums (u, a, n)
+        let bn_u_ptr = question_mark!(
+            translate_type::<u64>(memory_mapping, u_addr, self.loader_id, true),
+            result
+        );
+        let bn_a_ptr = question_mark!(
+            translate_type::<u64>(memory_mapping, a_addr, self.loader_id, true),
+            result
+        );
+        let bn_n_ptr = question_mark!(
+            translate_type::<u64>(memory_mapping, n_addr, self.loader_id, true),
+            result
+        );
+        let hg_result_address = question_mark!(
+            translate_type_mut::<u64>(memory_mapping, hg_result_addr, self.loader_id, true),
+            result
+        );
+        let nonce_len = unsafe { *(hg_result_address as *mut u64) };
+        let nonce_ptr = question_mark!(
+            translate_slice::<u8>(memory_mapping, nonce_addr, nonce_len, self.loader_id, true),
+            result
+        );
 
-        let account_infos = translate_slice::<SolAccountInfo>(
-            memory_mapping,
-            account_infos_addr,
-            account_infos_len,
-            self.loader_id,
-            enforce_aligned_host_addrs,
-        )?;
-        check_account_infos(account_infos.len(), &invoke_context)?;
-        let account_info_keys = account_infos
-            .iter()
-            .map(|account_info| {
-                translate_type::<Pubkey>(
-                    memory_mapping,
-                    account_info.key_addr,
-                    self.loader_id,
-                    enforce_aligned_host_addrs,
-                )
-            })
-            .collect::<Result<Vec<_>, EbpfError<BpfError>>>()?;
+        let u_bn = unsafe { &*(*bn_u_ptr as *mut BigNum) };
+        let a_bn = unsafe { &*(*bn_a_ptr as *mut BigNum) };
+        let n_bn = unsafe { &*(*bn_n_ptr as *mut BigNum) };
+        let bytes = (u_bn.num_bytes() + a_bn.num_bytes() + n_bn.num_bytes()) as f64;
+        question_mark!(
+            self.compute_meter
+                .consume(self.cost + calc_bignum_cost!(bytes)),
+            result
+        );
 
-        let translate = |account_info: &SolAccountInfo,
-                         invoke_context: &Ref<&mut dyn InvokeContext>| {
-            // Translate the account from user space
+        // hash_generator
+        let mut transcript = u_bn.as_ref().to_vec();
+        transcript.append(&mut a_bn.as_ref().to_vec());
+        // transcript.extend_from_slice(nonce_ptr.as_ref());
+        transcript.extend_from_slice(nonce_ptr);
+        // println!("t:{:?}",transcript);
+        let length = n_bn.num_bytes();
+        // println!("len:{}",length);
+        let h = Hkdf::<blake3::Hasher>::new(
+            Some(b"SST_SALT_HASH_TO_GENERATOR_"),
```

Comment:
The salt needs to be an input argument to the hash operation.

Reply:
Removed `blake3 hash` from PR... see 01ed0b9d229749506e0065fed170cac1310741b1